examples/neighbors/approximate_nearest_neighbors.py

```diff
@@ -121,191 +100,209 @@ def transform(self, X):
+        # plot TSNE embedding which should be very similar across methods
+        axes[i_ax].set_title(transformer_name + "\non " + dataset_name)
+        axes[i_ax].scatter(
+            Xt[:, 0],
+            Xt[:, 1],
+            c=y.astype(np.int32),
+            alpha=0.2,
+            cmap=plt.cm.viridis,
+        )
+        axes[i_ax].xaxis.set_major_formatter(NullFormatter())
+        axes[i_ax].yaxis.set_major_formatter(NullFormatter())
+        axes[i_ax].axis("tight")
+        i_ax += 1
 
-        msg = "Benchmarking on %s:" % dataset_name
-        print("\n%s\n%s" % (msg, "-" * len(msg)))
+fig.tight_layout()
+plt.show()
 
-        for transformer_name, transformer in transformers:
-            start = time.time()
-            Xt = transformer.fit_transform(X)
-            duration = time.time() - start
-
-            # print the duration report
-            longest = np.max([len(name) for name, model in transformers])
-            whitespaces = " " * (longest - len(transformer_name))
-            print("%s: %s%.3f sec" % (transformer_name, whitespaces, duration))
-
-            # plot TSNE embedding which should be very similar across methods
-            if "TSNE" in transformer_name:
-                axes[i_ax].set_title(transformer_name + "\non " + dataset_name)
-                axes[i_ax].scatter(
-                    Xt[:, 0],
-                    Xt[:, 1],
-                    c=y.astype(np.int32),
-                    alpha=0.2,
-                    cmap=plt.cm.viridis,
-                )
-                axes[i_ax].xaxis.set_major_formatter(NullFormatter())
-                axes[i_ax].yaxis.set_major_formatter(NullFormatter())
-                axes[i_ax].axis("tight")
-                i_ax += 1
-
-    fig.tight_layout()
-    plt.show()
-
-
-if __name__ == "__main__":
-    test_transformers()
-    run_benchmark()
+# %%
+# Sample output::
+#
+#     Benchmarking on MNIST_10000:
+#     ----------------------------
+#     TSNE with KNeighborsTransformer:     20.111 sec (fit_transform)
+#     TSNE with NMSlibTransformer:         21.757 sec (fit_transform)
+#     TSNE with internal NearestNeighbors: 24.828 sec (fit_transform)
+#
+#     Benchmarking on MNIST_20000:
+#     ----------------------------
+#     TSNE with KNeighborsTransformer:     50.994 sec (fit_transform)
+#     TSNE with NMSlibTransformer:         43.536 sec (fit_transform)
+#     TSNE with internal NearestNeighbors: 51.955 sec (fit_transform)
+#
+# Notice that the prediction speed
+# :class:`~sklearn.neighbors.KNeighborsTransformer` was optimized in
+# scikit-learn 1.1 and therefore the total `fit_transform` time of approximate
+# methods is not necessarily lower than the exact
+# :class:`~sklearn.neighbors.KNeighborsTransformer` solution. The reason is that
+# computing the index takes time and can nullify the benefits obtained by the
+# approximation. Indeed, the gains with respect to the exact solution increase
+# with increasing number of samples.
```

Comment:
```suggestion
# We can observe that the default `TSNE` estimator with its internal
# :class:`~sklearn.neighbors.NearestNeighbors` implementation is roughly
# equivalent the `TSNE` pipeline with
# :class:`~sklearn.neighbors.KNeighborsTransformer` in terms of performance.
# This is expected because both pipelines rely internally on the same
# :class:`~sklearn.neighbors.NearestNeighbors`NearestNeighbors` implementation
# that performs exacts neighbors search. The approximate `NMSlibTransformer` is
# already slightly faster than exact search on the smallest dataset but this
# speed difference is expected to become more significant on datasets with a
# larger number of samples.
#
# Note however that not all approximate search methods are guaranteed to
# improve upon the speed of the default exact search method: indeed the exact
# search implementation has been significantly improved in scikit-learn 1.1.
# Furthermore, the brute-force exact search method does not require building an
# index at `fit` time. So, to get an overall performance improvement in the
# context of the `TSNE` pipeline, the gains of the approximate search at
# `transform` need to be larger than the extra time speed to build the
# approximate search index at `fit` time.
#
# Finally, the TSNE algorithm itself is also computationally intensive,
# irrespective of the nearest neighbors search. So speeding-up the nearest
# neighbors search step by a factor of 5 would not result in a speed up by a
# factor of 5 for the overall pipeline.
```